supervisor/docker/addon.py

```diff
@@ -99,9 +99,7 @@ def timeout(self) -> int:
     @property
     def version(self) -> AwesomeVersion:
         """Return version of Docker image."""
-        if self.addon.legacy:
-            return self.addon.version
```

Comment:
@pvizeli It's a bit concerning to me that we're proposing making what's currently considered the legacy approach the universal approach. What's the history here? Was there some flaw in this approach that forced us to move away from it?

Reply:
Discord solved